src/lib.rs

```diff
@@ -425,36 +434,37 @@ impl CoverallsReport {
 
     /// Send report to the coveralls.io directly. For coveralls hosted on other
     /// platforms see send_to_endpoint
-    pub fn send_to_coveralls(&mut self) -> Result<(), curl::Error> {
+    pub fn send_to_coveralls(&mut self) -> Result<(), reqwest::Error> {
         self.send_to_endpoint("https://coveralls.io/api/v1/jobs")
     }
 
     /// Sends coveralls report to the specified url
-    pub fn send_to_endpoint(&mut self, url: &str) -> Result<(), curl::Error> {
+    pub fn send_to_endpoint(&mut self, url: &str) -> Result<(), reqwest::Error> {
         let body = match serde_json::to_vec(&self) {
             Ok(body) => body,
             Err(e) => panic!("Error {}", e),
         };
 
         let body = deflate_bytes_gzip(&body);
-        self.handle.url(url).unwrap();
-        let mut form = Form::new();
-        form.part("json_file")
-            .content_type("gzip/json")
-            .buffer("report", body)
-            .add()
-            .unwrap();
-        self.handle.httppost(form).unwrap();
-        self.handle.perform()
+
+        let form = Form::new().part(
+            "json_file",
+            Part::bytes(body).mime_str("gzip/json")?.file_name("report"),
+        );
+
+        let response = self.client.post(url).multipart(form).send()?;
```

Comment:
Note that this is now a blocking call, so looping on the value of `upload_status` does not make sense anymore. 

If you really want non-blocking, we could launch the client in a thread to keep the old polling behavior. Or, we could make a larger refactoring and move to `async` functions for sending requests, also using the `async` reqwest API.

Reply:
I am perfectly fine with a blocking API - after all tarpaulin would only use the blocking API :smile: